lib/ruby_lsp/requests/folding_ranges.rb

```diff
@@ -192,33 +197,37 @@ def emit_partial_range
         @partial_range = nil
       end
 
-      sig { params(node: T.any(SyntaxTree::Call, SyntaxTree::CommandCall)).void }
+      sig { params(node: T.any(SyntaxTree::CallNode, SyntaxTree::CommandCall)).void }
       def add_call_range(node)
         receiver = T.let(node.receiver, SyntaxTree::Node)
         loop do
           case receiver
-          when SyntaxTree::Call
+          when SyntaxTree::CallNode
             visit(receiver.arguments)
             receiver = receiver.receiver
           when SyntaxTree::MethodAddBlock
             visit(receiver.block)
             receiver = receiver.call
 
-            if receiver.is_a?(SyntaxTree::Call) || receiver.is_a?(SyntaxTree::CommandCall)
+            if receiver.is_a?(SyntaxTree::CallNode) || receiver.is_a?(SyntaxTree::CommandCall)
               receiver = receiver.receiver
             end
           else
             break
           end
         end
 
-        add_lines_range(receiver.location.start_line, node.location.end_line - 1)
+        add_lines_range(receiver.location.start_line, node.location.end_line - 1) if receiver
 
         visit(node.arguments)
       end
 
-      sig { params(node: T.any(SyntaxTree::Def, SyntaxTree::Defs)).void }
+      sig { params(node: SyntaxTree::DefNode).void }
       def add_def_range(node)
+        # For an endless method with no arguments, `node.params` returns `nil` for Ruby 3.0, but a `Syntax::Params`
+        # for Ruby 3.1
+        return unless node.params
```

Comment:
Is this guaranteed to work for regular method definitions? Will `params` never be `nil`?

Reply:
Ruby 3.0:

```
ruby-lsp andyw8/vini/upgrade-to-syntax-tree-5 % chruby 3.0
ruby-lsp andyw8/vini/upgrade-to-syntax-tree-5 % irb -r syntax_tree
irb(main):001:0> SyntaxTree.parse("def foo = 1").child_nodes.first.child_nodes.first.params.class
=> NilClass
irb(main):002:0> SyntaxTree.parse("def foo; end").child_nodes.first.child_nodes.first.params.class
=> SyntaxTree::Params                                                                              
irb(main):005:0> SyntaxTree.parse("def foo; end").child_nodes.first.child_nodes.first.params.child_nodes
=> [nil, nil, nil]
irb(main):006:0>
```
Ruby 3.1:

```
ruby-lsp andyw8/vini/upgrade-to-syntax-tree-5 % chruby 3.1
ruby-lsp andyw8/vini/upgrade-to-syntax-tree-5 % irb -r syntax_tree
irb(main):001:0> SyntaxTree.parse("def foo = 1").child_nodes.first.child_nodes.first.params.class
=> SyntaxTree::Params
irb(main):002:0> SyntaxTree.parse("def foo; end").child_nodes.first.child_nodes.first.params.class
=> SyntaxTree::Params
irb(main):003:0> SyntaxTree.parse("def foo; end").child_nodes.first.child_nodes.first.params.child_nodes
=> [nil, nil, nil]
```